plugins/python/src/main/kotlin/io/paddle/plugin/python/dependencies/GlobalCacheRepository.kt

```diff
@@ -0,0 +1,80 @@
+package io.paddle.plugin.python.dependencies
+
+import com.intellij.util.io.exists
+import io.paddle.plugin.python.Config
+import io.paddle.plugin.python.extensions.Requirements
+import java.nio.file.Files
+import java.nio.file.Path
+
+/**
+ * A service class for managing Paddle's cache.
+ *
+ * @see Config.cacheDir
+ */
+object GlobalCacheRepository {
+    private val cachedPackages: MutableList<CachedPackage> = mutableListOf()
+
+    init {
+        Config.cacheDir.toFile().listFiles()?.forEach { packageDir ->
+            packageDir.listFiles()?.forEach { versionDir ->
+                val descriptor = Requirements.Descriptor(name = packageDir.name, version = versionDir.name)
+                cachedPackages.add(CachedPackage(descriptor, versionDir.toPath()))
+            }
+        }
+    }
+
+    fun hasCached(descriptor: Requirements.Descriptor) = cachedPackages.any { it.descriptor == descriptor && it.srcPath.exists() }
```

Comment:
Is there any design pattern for keeping such `cachedPackages` list consistent with the actual list of packages from the cache in the filesystem? Updating it each time when it's needed is too expensive, I guess, but the current approach assumes that we also need to check each package and run `File.exists()` every time.. :thinking:

Reply:
I would consider setting up a daemon that would reload the current state of the dependency graph each minute and use its data as a source of truth for the rest of the application.